src/coreclr/jit/gentree.h

```diff
@@ -4683,47 +4684,15 @@ struct GenTreeCall final : public GenTree
         return (gtCallMoreFlags & GTF_CALL_M_EXPANDED_EARLY) != 0;
     }
 
-    //-----------------------------------------------------------------------------------------
-    // GetIndirectionCellArgKind: Get the kind of indirection cell used by this call.
-    //
-    // Arguments:
-    //     None
-    //
-    // Return Value:
-    //     The kind (either R2RIndirectionCell or VirtualStubCell),
-    //     or NonStandardArgKind::None if this call does not have an indirection cell.
-    //
-    NonStandardArgKind GetIndirectionCellArgKind() const
-    {
-        if (IsVirtualStub())
-        {
-            return NonStandardArgKind::VirtualStubCell;
-        }
-
-#if defined(TARGET_ARMARCH)
-        // For ARM architectures, we always use an indirection cell for R2R calls.
-        if (IsR2RRelativeIndir())
-        {
-            return NonStandardArgKind::R2RIndirectionCell;
-        }
-#elif defined(TARGET_XARCH)
-        // On XARCH we disassemble it from callsite except for tailcalls that need indirection cell.
-        if (IsR2RRelativeIndir() && IsFastTailCall())
-        {
-            return NonStandardArgKind::R2RIndirectionCell;
-        }
-#endif
-
-        return NonStandardArgKind::None;
-    }
+    bool HasIndirectionCellArg(fgArgTabEntry** entry) const;
```

Comment:
Consider adding a default arg value so callers who don't care about the entry don't need to explicitly pass `nullptr`.

```suggestion
    bool HasIndirectionCellArg(fgArgTabEntry** entry = nullptr) const;
```